crypto/ethsecp256k1/ethsecp256k1.go

```diff
@@ -105,10 +104,14 @@ func (privKey *PrivKey) UnmarshalAminoJSON(bz []byte) error {
 }
 
 // Sign creates a recoverable ECDSA signature on the secp256k1 curve over the
-// Keccak256 hash of the provided message. The produced signature is 65 bytes
+// provided hash of the message. The produced signature is 65 bytes
 // where the last byte contains the recovery ID.
-func (privKey PrivKey) Sign(msg []byte) ([]byte, error) {
-	return ethcrypto.Sign(ethcrypto.Keccak256Hash(msg).Bytes(), privKey.ToECDSA())
+func (privKey PrivKey) Sign(digestBz []byte) ([]byte, error) {
+	if len(digestBz) != ethcrypto.DigestLength {
```

Comment:
Hello. This conditional signing makes it different from the previous codebase we had. And introduces discrepancy with many ethermint-based codebases e.g. https://github.com/informalsystems/ibc-rs/pull/1295#pullrequestreview-734631915

And actually in my opinion this can lead to very shady bugs when one expects always a keccak, but if input matches 32 bytes (and IS NOT keccak), it will implicitly be omitted. I never saw that in other projects (cosmos / ethereum related).

Reply:
Ack

Reply:
Guys? :)